app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  before_action :require_logged_out, only: [:new]
+
+  # TODO: message for require_logged_out should be suppressed
```

Comment:
Rather than requiring logged out, I prefer to redirect logged in users to some kind of dashboard (or root/home page), and setting a flash message telling them they are logged in.

I also prefer to show that there is a 'new' action, even tho for Rails it is not required. It is better when one looks at the controller and sees actually there is an action.

```ruby
# Like this
def new; end
```